public/pages/CreateMonitor/components/MonitorExpressions/expressions/WhereExpression.js

```diff
@@ -58,262 +27,117 @@ const propTypes = {
-              <EuiFlexGroup style={{ ...EXPRESSION_STYLE }}>
-                <EuiFlexItem grow={false} style={{ width: 200 }}>
-                  <FormikComboBox
-                    name={`${fieldPath}where.fieldName`}
-                    inputProps={{
-                      placeholder: 'Select a field',
-                      options: indexFields,
-                      onChange: this.handleFieldChange,
-                      isClearable: false,
-                      singleSelection: { asPlainText: true },
-                    }}
-                  />
-                </EuiFlexItem>
-                <EuiFlexItem grow={false}>
-                  <FormikSelect
-                    name={`${fieldPath}where.operator`}
-                    inputProps={{
-                      onChange: this.handleOperatorChange,
-                      options: fieldOperators,
-                    }}
-                  />
-                </EuiFlexItem>
-                {!isNullOperator(fieldOperator) && (
-                  <EuiFlexItem>{this.renderValueField(fieldType, fieldOperator)}</EuiFlexItem>
-                )}
-              </EuiFlexGroup>
-            </div>
-          </EuiPopover>
+          <div style={{ maxWidth: '75%' }}>
+            {filters.map((filter, index) => (
+              <span key={`${fieldPath}filters.${index}`} style={{ paddingRight: '5px' }}>
+                <WherePopover
+                  closePopover={() => closeExpression(Expressions.WHERE)}
+                  dataTypes={dataTypes}
+                  indexFieldFilters={indexFieldFilters}
+                  fieldPath={`${fieldPath}filters.${index}.`}
+                  filtersArrayHelpers={filtersArrayHelpers}
+                  formik={formik}
+                  index={index}
+                  onMadeChanges={onMadeChanges}
+                  openPopover={() => openExpression(Expressions.WHERE)}
+                  openedState={openedStates.WHERE}
+                  useTriggerFieldOperators={useTriggerFieldOperators}
+                  values={filter}
+                  whereFilterHeader={whereFilterHeader}
+                />
+              </span>
+            ))}
+          </div>
         )}
 
-        {showAddButtonFlag && (
+        <EuiSpacer size={'xs'} />
+        {filters.length < maxNumWhereExpressions && (
           <EuiButtonEmpty
             size="xs"
-            data-test-subj={`${fieldPath}where.addFilterButton`}
-            onClick={() => openExpression(Expressions.WHERE)}
+            data-test-subj={`${fieldPath}addFilterButton`}
+            onClick={() => {
+              openExpression(Expressions.WHERE);
+              filtersArrayHelpers.push(
+                useTriggerFieldOperators
+                  ? {
+                      ...FORMIK_INITIAL_WHERE_EXPRESSION_VALUES,
+                      operator: TRIGGER_OPERATORS_MAP.INCLUDE,
+                    }
+                  : FORMIK_INITIAL_WHERE_EXPRESSION_VALUES
```

Comment:
Should this still be a new object?
`{ ...FORMIK_INITIAL_WHERE_EXPRESSION_VALUES }`

Reply:
Good point. Revised.